timesketch/frontend-ng/src/components/Explore/AddManualEvent.vue

```diff
@@ -0,0 +1,182 @@
+          </v-text-field>
+        </v-col>
+      </v-row>
+
+      <v-row v-for="(attribute, index) in attributes" :key="index">
+        <v-col cols="6">
+          <v-text-field
+            label="Attribute Name"
+            outlined
+            hide-details
+            v-model="attributes[index].name"
+            @change="setAddAttributeBtn(index)"
+          >
+          </v-text-field>
+        </v-col>
+        <v-col cols="6">
+          <v-text-field
+            label="Attribute Value"
+            outlined
+            hide-details
+            v-model="attributes[index].value"
+            @change="setAddAttributeBtn(index)"
+          >
+          </v-text-field>
+        </v-col>
+      </v-row>
+
+      <v-card-actions>
+        <v-spacer></v-spacer>
+        <v-btn text color="primary" @click="addAttribute" :disabled="isDisabled"> Add Attribute </v-btn>
+        <v-btn text color="primary" @click="clearAndCancel"> Cancel </v-btn>
+        <v-btn text color="primary" @click="submit"> Add Event </v-btn>
+      </v-card-actions>
+    </v-container>
+  </v-card>
+</template>
+
+<script>
+import dayjs from '@/plugins/dayjs'
+import DatePicker from 'v-calendar/lib/components/date-picker.umd'
+import ApiClient from '../../utils/RestApiClient'
+
+export default {
+  props: ['datetimeProp'],
+  components: {
+    DatePicker,
+  },
+  data() {
+    return {
+      message: null,
+      timestampDesc: null,
+      filterTab: null,
+      showPicker: false,
+      attributes: [],
+      isDisabled: false,
+      datetime: this.datetimeProp,
+    }
+  },
+  watch: {
+    datetimeProp: function () {
+      this.datetime = this.datetimeProp
+    },
+  },
+  computed: {
+    dateFromPicker: {
+      set(val) {
+        if (val !== null) this.datetime = dayjs.utc(val).millisecond(0).toISOString()
+      },
+      get() {
+        return this.datetime
+      },
+    },
+  },
+  methods: {
+    setAddAttributeBtn: function (i) {
+      if (this.attributes[i].name === '' || this.attributes[i].value === '') {
+        this.isDisabled = true
+      } else {
+        this.isDisabled = false
+      }
+    },
+    addAttribute: function () {
+      this.attributes.push({ name: '', value: '' })
+      this.isDisabled = true
+    },
+    clearAndCancel: function () {
+      this.datetime = null
+      this.message = null
+      this.timestampDesc = null
+      this.attributes = []
+      this.isDisabled = false
+      this.$emit('cancel')
+    },
+    submit: function () {
+      if (this.datetime === null || this.message === null || this.timestampDesc === null) {
+        return
+      }
+      let sketchId = this.$store.state.sketch.id
+      let config = {
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }
+      let attributes = {}
+      this.attributes.forEach((attribute) => {
+        if (attribute.name === '' || attribute.value === '') return
+        attributes[attribute.name] = attribute.value
+      })
```

Comment:
It's probably fine to leave this this way, but I'd change this to use `filter` and `reduce` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce).

Reply:
```
<v-btn text color="primary" @click="attributes.push({ name: '', value: '' })" :disabled="isDisabled"> Add Attribute </v-btn>

```

Reply:
I am not sure I understood: do you mean something like this?
```
let attributes = {}
this.attributes.reduce((_, attribute) => {
  if (!(attribute.name === '' || attribute.value === '')) attributes[attribute.name] = attribute.value
}, attributes)
```